frontend/src/App.jsx

```diff
@@ -54,55 +61,57 @@ export default function App() {
         <AlertContext.Provider value={simpleSetAlert}>
           <UserContext.Provider value={user}>
             <BiolinkContext.Provider value={biolink}>
-              <ThemeProvider theme={theme}>
-                <StylesProvider injectFirst>
-                  <AlertWrapper
-                    alert={alert}
-                    onClose={() => simpleSetAlert(alert.severity, '')}
-                  />
-                  <Header user={user} setUser={setUser} />
-                  <div id="contentContainer">
-                    <Switch>
-                      <Route path="/about">
-                        <About />
-                      </Route>
-                      <Route path="/help">
-                        <Help />
-                      </Route>
-                      <Route path="/guide">
-                        <Guide isSignedIn={isSignedIn} />
-                      </Route>
-                      <Route path="/neighborhood">
-                        <Neighborhood />
-                      </Route>
-                      <Route path="/questions">
-                        <QuestionList />
-                      </Route>
-                      <Route path="/question/:question_id">
-                        <QuestionAnswerViewer />
-                      </Route>
-                      <Route path="/termsofservice">
-                        <TermsofService />
-                      </Route>
-                      <Route path="/simple/view">
-                        <SimpleViewer
-                          user={user}
-                        />
-                      </Route>
-                      <Route path="/simple/question">
-                        <SimpleQuestion />
-                      </Route>
-                      <Route path="/q/new">
-                        <QuestionNew />
-                      </Route>
-                      <Route path="/">
-                        <Landing isSignedIn={isSignedIn} />
-                      </Route>
-                    </Switch>
-                  </div>
-                  <Footer />
-                </StylesProvider>
-              </ThemeProvider>
+              <ConceptsContext.Provider value={concepts}>
```

Comment:
Context providers are kinda ugly and I think we should try to minimize their use as much as possible. We could get the same information as the ConceptsContext using the BiolinkContext. I think it's cleaner to just pull that information from the BiolinkContext whenever we need it rather than creating a new context for Concepts because the concepts exist in the biolink context.

This can be done in only a couple lines:
```javascript
const biolink = useContext(BiolinkContext);
const concepts = biolinkUtils.getValidConcepts(biolink);
```

Reply:
We use concepts in many places, whereas we only use one other property of biolink in the edge panel. I would like to keep the concepts context for these reasons:
- it might be extra in App, but then we'd have to add an extra line in at least 8 different places
- we'd only be importing a small-ish array instead of the whole biolink object
- We do the getValidConcepts operation once to get a static array instead of in a bunch of different components, and then we might need to memoize those so we don't do the operation on every new render.